### PR TITLE
Consolidate http status assertion in E2Es

### PR DESCRIPTION
**Pull request by tamirkamara**
## What is being addressed

On some of the E2E tests an assertion has been made over the http status code without logging the message that came with it. This made it hard to understand the exact error.

## How is this addressed

- Build a single method to handle all http status checks that will also print out the text in case of an error.

### CODE REVIEW

**Review by eladiw — APPROVED** [review]

File: e2e_tests/test_workspace_templates.py [hunk 1]
```diff
@@ -33,4 +33,4 @@ async def test_get_workspace_templates(template_name, verify) -> None:
 async def test_get_workspace_template(template_name, verify) -> None:
     admin_token = await get_admin_token(verify)
     async with get_template(template_name, strings.API_WORKSPACE_TEMPLATES, admin_token, verify) as response:
-        assert (response.status_code == status.HTTP_200_OK), f"GET Request for {template_name} creation failed"
+        assert_status(response, [status.HTTP_200_OK], f"Failed to create template: {template_name}")
```
Comment (yuvalyaron) [inline e2e_tests/test_workspace_templates.py:36]: does it necessarily apply that the creation has failed?

**Review by yuvalyaron — CHANGES_REQUESTED** [review]

File: e2e_tests/test_workspace_templates.py (hunk 1 above)
Comment (tamirkamara) [reply e2e_tests/test_workspace_templates.py:36]: @yuvalyaron what do you mean?

Comment (yuvalyaron) [reply e2e_tests/test_workspace_templates.py:36]: if we fail to get the workspace template - we see a message that says that the creation of the template has failed,
I'm asking if there are other options. 
i.e. the creation was ok, but there's a problem with the token, or with the endpoint.
Is it ok to assume that it's the creation that failed?

Comment (yuvalyaron) [reply e2e_tests/test_workspace_templates.py:36]: because the test is not creating the template, it just calls GET on it

Comment (tamirkamara) [reply e2e_tests/test_workspace_templates.py:36]: I made a mistake in the error message. it should be:
```suggestion
        assert_status(response, [status.HTTP_200_OK], f"Failed to GET template: {template_name}")
```

**Review by yuvalyaron — APPROVED** [review]

**Pull request merged**